src/ol/uri.js

```diff
@@ -0,0 +1,17 @@
+goog.provide('ol.uri');
+
+
+/**
+ * Appends query parameters to a URI.
+ *
+ * @param {string} uri The original URI, which may already have query data.
+ * @param {!Object} params An object where keys are URI-encoded parameter keys,
+ *     and the values are arbitrary types or arrays.
+ * @return {string} The new URI.
+ */
+ol.uri.appendParams = function(uri, params) {
+  var qs = Object.keys(params).map(function(k) {
+    return encodeURIComponent(k) + '=' + encodeURIComponent(params[k]);
+  });
+  return uri + (uri.indexOf('?') === -1 ? '?' : '&') + qs.join('&');
```

Comment:
I think you'll have to leave examples about what you think should be supported to make it clearer.  These can then be added to the tests below.